src/test/java/BaseTest.java

```diff
@@ -1,11 +1,211 @@
 import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.safari.SafariDriver;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Parameters;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import java.net.MalformedURLException;
+import java.net.URI;
+
+import javax.swing.*;
+import java.time.Duration;
 
 public class BaseTest {
+    public static WebDriver driver = null;
+
+    public static String url = "https://qa.koel.app/";
 
+    public static WebDriverWait wait = null;
+
+    public static Actions actions = null;
     @BeforeSuite
     static void setupClass() {
-        WebDriverManager.chromedriver().setup();
+
+//        WebDriverManager.chromedriver().setup();
+//        WebDriverManager.FirefoxDriver().setup();
+
+
+    }
+
+    @BeforeMethod
+    @Parameters ({"baseUrl"})
+    public void lunchBrowser(String baseUrl) {
+
+        //      Added ChromeOptions argument below to fix websocket error
+
+//        ChromeOptions options = new ChromeOptions();
+//        options.addArguments("--remote-allow-origins=*");
+//        driver = new FirefoxDriver();
+        driver = pickBrowser(System.getProperty("browser"));
```

Comment:
The code is not compiled! Please finish the homework!You got this problem, please correct it. 

![image](https://github.com/Alexer0105/A47-QA-Automation-Framework/assets/104761482/c3c90afe-ddbc-4d7f-a13d-644801632fa0)

HINT

![image](https://github.com/Alexer0105/A47-QA-Automation-Framework/assets/104761482/95ecdf09-500c-4c77-a719-1a9e0d2a6edd)